contracts/root/RootERC20Predicate.sol

```diff
@@ -71,22 +75,49 @@ contract RootERC20Predicate is Initializable, IRootERC20Predicate {
      * @inheritdoc IRootERC20Predicate
      */
     function deposit(IERC20Metadata rootToken, uint256 amount) external {
+        uint256 expectedBalance = rootToken.balanceOf(address(this)) + amount;
         _deposit(rootToken, msg.sender, amount);
+        // invariant check to ensure that the root token balance has increased by the amount deposited
+        require((rootToken.balanceOf(address(this)) == expectedBalance), "RootERC20Predicate: UNEXPECTED_BALANCE");
     }
 
     /**
      * @inheritdoc IRootERC20Predicate
      */
     function depositTo(IERC20Metadata rootToken, address receiver, uint256 amount) external {
+        uint256 expectedBalance = rootToken.balanceOf(address(this)) + amount;
         _deposit(rootToken, receiver, amount);
+        // invariant check to ensure that the root token balance has increased by the amount deposited
+        require((rootToken.balanceOf(address(this)) == expectedBalance), "RootERC20Predicate: UNEXPECTED_BALANCE");
+    }
+
+    /**
+     * @inheritdoc IRootERC20Predicate
+     */
+    function depositNativeTo(address receiver) external payable {
+        require(msg.value > 0, "RootERC20Predicate: INVALID_AMOUNT");
+        _deposit(IERC20Metadata(NATIVE_TOKEN), receiver, msg.value);
     }
 
     /**
      * @inheritdoc IRootERC20Predicate
      */
     function mapToken(IERC20Metadata rootToken) public returns (address) {
         require(address(rootToken) != address(0), "RootERC20Predicate: INVALID_TOKEN");
-        require(rootTokenToChildToken[address(rootToken)] == address(0), "RootERC20Predicate: ALREADY_MAPPED");
+        return _map(address(rootToken), rootToken.name(), rootToken.symbol(), rootToken.decimals());
+    }
+
+    function _mapNative() private {
+        _map(NATIVE_TOKEN, "Ether", "ETH", 18);
+    }
+
+    function _map(
+        address rootToken,
+        string memory tokenName,
+        string memory tokenSymbol,
```

Comment:
@drinkcoffee isn't the use of `memory` redundant?

Reply:
The location of the string needs to be specified: storage, memory or calldata